File: app/models/like.py
from sqlalchemy import Column, Integer, String, Boolean, DateTime
from sqlalchemy.sql import func
from ..database import Base

class Like(Base):
    __tablename__ = "t_app_like"
    
    id = Column(Integer, primary_key=True, index=True)
    user = Column(String(100), nullable=False)
    product = Column(String(100), nullable=False)
    is_favorite = Column(Boolean, nullable=False)
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    
    def __str__(self):
        return f"{self.user} - {self.product} - {self.is_favorite}" 

File: models/Like.js
const { DataTypes } = require('sequelize');
const { sequelize } = require('../config/database');
const User = require('./User');
const Product = require('./Product');

const Like = sequelize.define('Like', {
  id: {
    type: DataTypes.INTEGER,
    primaryKey: true,
    autoIncrement: true
  },
  user_id: {
    type: DataTypes.INTEGER,
    allowNull: false,
    references: {
      model: 'auth_user',
      key: 'id'
    }
  },
  product_id: {
    type: DataTypes.BIGINT,
    allowNull: false,
    references: {
      model: 't_app_product_product',
      key: 'id'
    }
  },
  created_at: {
    type: DataTypes.DATE,
    defaultValue: DataTypes.NOW
  }
}, {
  tableName: 'likes',
  timestamps: false,
  indexes: [
    {
      unique: true,
      fields: ['user_id', 'product_id']
    }
  ]
});

// Associations
Like.belongsTo(User, { foreignKey: 'user_id', as: 'user' });
Like.belongsTo(Product, { foreignKey: 'product_id', as: 'product' });
User.hasMany(Like, { foreignKey: 'user_id', as: 'likes' });
Product.hasMany(Like, { foreignKey: 'product_id', as: 'likes' });

module.exports = Like; 